tests/test_live.py

```diff
@@ -0,0 +1,340 @@
+        shape: Figure
+        align_at: str
+        rect: Rect
+
+        @property
+        def align(self):
+            if self.align_at == "b_box":
+                return self.shape.b_box
+            elif self.align_at == "pr_boundary":
+                return self.shape.pr_boundary
+
+        @property
+        def b_box(self) -> BoundingBox:
+            if self.align_at == "b_box":
+                return self.shape.skill_b_box
+            elif self.align_at == "pr_boundary":
+                return self.shape.skill_pr_boundary
+
+    count = 0
+
+    @contextmanager
+    def create_inst():
+        nonlocal count, ws, cv, cell_cv
+        name = f"I{count}_test"
+        db = ws.db.create_inst(cv, cell_cv, name, (0, 0), "R0")
+        count += 1
+        try:
+            shape = Instance.from_name(cv, name)
+            yield Alignable(shape=shape, align_at="b_box", rect=Rect[-1:11, -3:7])
+        finally:
+            ws.db.delete_object(db)
+
+    @contextmanager
+    def create_pr_inst():
+        # Instance to be aligned at the P&R boundary
+        with create_inst() as inst:
+            yield Alignable(shape=inst.shape, align_at="pr_boundary", rect=Rect[-1:7, -3:5])
+
+    @contextmanager
+    def create_inst_group():
+        with create_inst() as i1:
+            with create_inst() as i2:
+                shape = i1.shape.b_box.align(center_left=i2.shape.b_box.center_right)
+                b_box = shape.skill_b_box
+                yield Alignable(
+                    shape=shape,
+                    align_at="b_box",
+                    rect=Rect[b_box[0][0] : b_box[1][0], b_box[0][1] : b_box[1][1]],
+                )
+
+    @contextmanager
+    def create_db_shape():
+        nonlocal cv
+        canvas = Canvas(cv)
+        rect = Rect[0:1, 3:7, dummy_layer]
+        group = Group([rect])
+        canvas.append(group)
+        (db,) = canvas.draw()
+
+        rect_db = DbShape(db.db)
+
+        yield Alignable(shape=rect_db, align_at="b_box", rect=rect)
+        rect_db.delete()
+
+    @contextmanager
+    def create_rod_shape():
+        nonlocal ws, cv
+        b_box = ((-3, -5), (13, 17))
+        rod = ws.rod.create_rect(cv_id=cv, layer=dummy_layer, b_box=b_box)
+        shape = RodShape.from_rod(cast(RemoteObject, rod))
+        yield Alignable(
+            shape=shape,
+            align_at="b_box",
+            rect=Rect[b_box[0][0] : b_box[1][0], b_box[0][1] : b_box[1][1]],
+        )
+        shape.delete()
+
+    # Align for bounding box
+    handles = python_skill_handle
+    shapes = (create_inst, create_pr_inst, create_inst_group, create_db_shape, create_rod_shape)
+    for maintain in (True, False):
+        for handle1, handle2 in ((x, y) for x in handles for y in handles):
+            for create_s1, create_s2 in ((x, y) for x in shapes for y in shapes):
+                with create_s1() as s1:
+                    with create_s2() as s2:
+                        if maintain and (
```

Comment:
This should be a [parametrized test case](https://docs.pytest.org/en/latest/parametrize.html) as it currently look

Also, this function is way too long